### PR TITLE
Cmap format4 fixes

### PR DESCRIPTION
**Pull request by readroberts**
Adopt logic from fontTools for building a format 4 cmap. This produces a smaller cmap subtable size.

### CODE REVIEW

File: afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/cmap.c [hunk 1]
```diff
@@ -719,64 +719,125 @@ static Format2 *makeFormat2(cmapCtx h) {
...
 /* ------- end format 2 */
 
+static void setOrdered( Mapping *mapping, int codeBreak, int span, int index)
+{
+    int threshold;
+    /* If the ordered segment starts with a code break or ends with a code break,
+     then the cost of keeping this as a separate segment is just one new segment break
+     (the following segment break).
+     Else, the cost is adding two new segment breaks - the one to start this segment,
+     and to start the next segment. We mark as NOT orderd those segments that are no larger
```
Comment (cjchapman) [inline afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/cmap.c]: typo: `orderd` -> `ordered`

File: afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/cmap.c [hunk 2]
```diff
@@ -719,64 +719,125 @@ static Format2 *makeFormat2(cmapCtx h) {
...
+        mapping[index - span].ordered = 1;
+
+}
+
 /* Partition mappings optimally for format 4. Assumes mapping in code order */
 static int partitionRanges(cmapCtx h, Mapping *mapping) {
 	int span;
 	int ordered;
-	int overflow;
+    int lastOrdered;
```
Comment (cjchapman) [inline afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/cmap.c]: seems to be a mix of tabs vs spaces here, please make the indentation consistent

File: afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/cmap.c [hunk 3]
```diff
@@ -719,64 +719,125 @@ static Format2 *makeFormat2(cmapCtx h) {
...
+            mapping[i - span].ordered = 0; /* default */
+            if (span > 1) /* an ordered segment is a segment with more than one code point with consecutive glyph ID's */
+            {
+                setOrdered(mapping, codeBreak, span, i);
+            }
+            if (codeBreak) {
+                mapping[i].flags |= CODE_BREAK;
+            }
 			span = 1;
+            nSegments++;
```
Comment (cjchapman) [inline afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/cmap.c]: seems to be another mix of tabs vs spaces here, please make the indentation consistent

File: afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/cmap.c [hunk 4]
```diff
@@ -843,8 +902,6 @@ static Format4 *makeFormat4(cmapCtx h, unsigned long *length) {
         truncateCmap = 1;
 
 	while (notOK) {
-		unsigned int numOrdered = 0;
-		unsigned int numNotOrdered = 0;
 
 		fmt->glyphId.cnt = 0;
         if (h->mapping.cnt == 0)
```
Comment (cjchapman) [inline afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/cmap.c]: seems to be another mix of tabs vs spaces here, please make the indentation consistent

File: afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/cmap.c (hunk 3 above)
Comment (cjchapman) [inline afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/cmap.c]: Codacy points out that nSegments is uninitialized the first time through.

**Comment by readroberts** [comment]
Closing pull requests so I can delete the branch

**Pull request closed without merging**